### ISSUE TITLE
Feature request: Tabs in excalidraw

### ISSUE THREAD

**Issue by rajikaimal** [excalidraw/excalidraw#1543]
Usually I need to work on multiple diagrams at the same time. Tabs should be a good way to work on different workspaces at the same time.

**Comment by kbariotis** [comment]
> Usually I need to work on multiple diagrams at the same time.

@rajikaimal could you please elaborate on this? Would you like to be able to view them at the same time? Some sort of split-screen? Or have shortcuts that you could jump from one diagram to another?

**Comment by crash7** [comment]
I was about to request the same, maybe we can do something like this namespacing the state it saves into localStorage, so you can switch between diagrams without doing a "load" (keeping in mind we could hit a localStorage limit with many diagrams).

Basically, in my case (not saying that it's the same as @rajikaimal), I would like to jump between recent diagrams without doing a load (because since they are saved as untitled, I end with a downloads folder with a lot of "untitled" files, haha)

**Comment by lipis** [comment]
> because since they are saved as untitled, I end with a downloads folder with a lot of "untitled" files, haha

You can change the name from Export dialog :)

Maybe we should prompt for a name the first time we try to download the file if it's `untitled`..

**Comment by lipis** [comment]
> @rajikaimal could you please elaborate on this?

@kbariotis Think of them like the sheets in the Excel :)

**Comment by rajikaimal** [comment]
> could you please elaborate on this? Would you like to be able to view them at the same time? Some sort of split-screen? Or have shortcuts that you could jump from one diagram to another?

I needed to work on two diagrams at the same time, so pretty much jumping back and forward between 2 diagrams. Having a split screen feature would be cool too, but I meant a mechanism similar to browser tabs.

**Comment by rrikesh** [comment]
If you're using Firefox, use [containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/) as a quick workaround.

**Comment by Nopik** [comment]
So, I'm a casual user which occasionally draws some small diagrams in the tool. I do like that whenever I go to excalidraw, it basically shows what I was drawing first. I do not really care if that is local storage or server side. But, I want to keep that diagrams handy in the tool (not in a tons of local json files), and still be able to draw new things. So, I ended up having one giant diagram with all I drawn so far over past few weeks - and I'm just scrolling around every time I need something. So I can easily copy elements from old pictures (I do reuse elements a lot).
Having some way to organize that - again, in the tool, not as bunch of files I can load one at a time - would be my #1 (and probably the only) wishlist item. Any tabs, file organizer, whatever.

**Comment by Fausto95** [comment]
In my opinion what we really need here is Artboards

**Comment by vjeux** [comment]
I like the gallery concept of Procreate
https://youtu.be/EI-ToB0tG9A

**Comment by tomayac** [comment]
I’m happy to work on this, since on the Chrome team we’re working on adding native tabbed mode. Excalidraw would be a great example to showcase this feature.

**Comment by lipis** [comment]
Go for it!!

**Comment by tomayac** [comment]
We have discussed this on the team chat and came to the conclusion that there are two feature requests. I have opened #2597 for the feature request I have.

**Comment by tanishqmanuja** [comment]
Recently started using Obsidian with Excalidraw plugin. It's not perfect but does the job of working with/managing multiple drawings at the same time.

**Comment by tanishqmanuja** [comment]
Another app in the same category has a concept of pages, ref: https://github.com/tldraw/tldraw

**Comment by Atreyu-94** [comment]
@tanishqmanuja They did really well with the pages.

**Comment by tanishqmanuja** [comment]
Yeah they really nailed down some features like multiplayer, pages and the component layering & anchoring stuff. The creator also explained his thought process in a YT video. I still like excalidraw ... that's why i think sheet/pages are a must, idk why it's taking so long to implement?

**Comment by tomayac** [comment]
Storage would have to be rethought. Right now, it's all in `localStorage.excalidraw`. With tabs, there would need to be some name-spaced storage mechanism with on storage bucket per tab.

**Comment by tanishqmanuja** [comment]
Hmm that was not hard to guess, But the issue's age is more than 3 years now. It could have been done, many approaches are there like Document-Page model similar to what tldraw uses or maybe simpler like saving pages to an indexeddb store, and only load the active page to localStorage.excalidraw.

PS:
This is not intended to be read as some hyped criticism, My intension is only the bring attention to the pain point which I experience regularly because this is my go to tool for explaining concepts to fellow mates, and I don't wanna open incognito or multiple browsers just to keep the context different for each presentation.

Excalidraw is still more feature rich than tldraw as of now, It will be great if minor details are reworked :)

**Comment by eduardolat** [comment]
Really needed

**Comment by alexk111** [comment]
To have multiple Excalidraw instances in right contexts, I use it in Freeter (https://github.com/FreeterApp/Freeter). 

Example of a "work on two diagrams at the same time" workflow:

![image](https://github.com/excalidraw/excalidraw/assets/4675984/3121b37f-f194-40b6-9e63-cfd707248094)

Example of a workflow for brainstorming feature ideas:

![image](https://github.com/excalidraw/excalidraw/assets/4675984/32fae749-aedc-4c19-9424-9208cdc2b39f)

**Comment by MontejoJorge** [comment]
I just made this [https://github.com/MontejoJorge/excalidraw-multi-tabs](url), its not perfect, but it works

**Comment by yaner-here** [comment]
It seems that the tabs feature will never be implemented, since the developers said that it won't be considered. There's a similar conception called `scene` in their payment-required ExcaliDraw Plus, which allows you to render multiple ExcaliDraw subgraph in a single ExcaliDraw graph.

> (https://github.com/excalidraw/excalidraw/issues/9596#issuecomment-2922623255) Hey, we have a task for the pages/tabs, but there is no plan to add those in a near future, since [plus.excalidraw.com](https://plus.excalidraw.com/) already allows to have multiple scenes.

Sign, what a pity.

**Comment by florihupf** [comment]
Just make sure that `isBrowserStorageStateNewer` returns false.

**Comment by 1ucas** [comment]
I'll open a Pull Request by Monday on this.

But you guys can test this feature here: https://1ucas.github.io/excalitabs/

This fork is basically 1 commit over this repo's master branch adding multi-tab support.

Feel free to throw in some feedbacks.

**Comment by bvleur** [comment]
@1ucas: Great start! I would love to see this.

- Undo/redo seems broken when switching tabs.
- I personally think tabs should be top or "inline start" (left//right depending on locale). Bottom is a bit hard to discover.

**Comment by 1ucas** [comment]
Thanks @bvleur ! 

We're trying to bring the feature as light as possible, without disrupting the folder capability for subscribers.

But it still got some rough edges. 
So we made this temp "Excalitabs" demo to cover as many scenarios as possible before open the PR for review. 

There are also some things worth discussing at the ui like tab position and tab selection.

Thanks for the feedback!

**Comment by 1ucas** [comment]
The reason for the bottom position is that it fits the behavior of some drawing tools we're used to. 

Maybe we should give it a more prominent visual?
